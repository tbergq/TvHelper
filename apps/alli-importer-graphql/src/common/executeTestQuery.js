// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

const createContext = () => ({});

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
