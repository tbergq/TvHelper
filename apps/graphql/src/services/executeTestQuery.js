// @flow

import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
