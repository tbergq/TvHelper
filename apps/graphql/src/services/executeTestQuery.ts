import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: {} | null | undefined,
  request: {} | null | undefined = {},
): Promise<{}> {
  // @ts-ignore: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
