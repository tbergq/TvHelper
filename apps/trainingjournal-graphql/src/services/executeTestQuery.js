// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

export default function executeTestQuery(query: string, variables: ?{ ... }): Promise<{ ... }> {
  return graphql(schema, query, null, null, variables);
}
