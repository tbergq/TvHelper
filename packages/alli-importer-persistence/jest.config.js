// @flow

const path = require('path');
const { findMonorepoRoot } = require('@adeira/monorepo-utils');

const setTestPath = (path.join(findMonorepoRoot(), 'scripts', 'setupTest.js') /*: string */);
const mongoTestPath = (path.join(__dirname, 'scripts', 'mongoMemoryServer.js') /*: string */);

module.exports = {
  rootDir: __dirname,
  testEnvironment: 'node',
  setupFilesAfterEnv: [setTestPath, mongoTestPath],
};
