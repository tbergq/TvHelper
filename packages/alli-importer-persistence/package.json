{
  "name": "@tbergq/alli-importer-persistence",
  "private": true,
  "version": "0.0.1",
  "main": "src/index",
  "sideEffects": false,
  "dependencies": {
    "@tbergq/persistence-utils": "0.0.1",
    "mongoose": "^5.9.27"
  },
  "devDependencies": {
    "@adeira/monorepo-utils": "^0.8.0",
    "mockdate": "^3.0.2",
    "mongodb-memory-server": "^6.6.3"
  }
}
