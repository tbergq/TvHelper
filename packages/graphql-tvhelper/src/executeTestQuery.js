// @flow

import { graphql as originalGraphQL, GraphQLSchema, GraphQLObjectType } from 'graphql';

import SearchTvShow from './search/queries/SearchTvShow';

const RootQuery = new GraphQLObjectType({
  name: 'RootQuery',
  description: 'Root Query',
  fields: {
    searchTvShow: SearchTvShow,
  },
});

const schema = new GraphQLSchema({
  query: RootQuery,
});

/**
 * Executes GraphQL query using our schema. This function should
 * be used only during testing because it uses faked context.
 */
export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  context: ?{ ... },
): Promise<Object> {
  return originalGraphQL(schema, query, null, context, variables);
}
