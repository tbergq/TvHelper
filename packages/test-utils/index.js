// @flow

export { default as withMongoDb } from './src/useMongoDb';
