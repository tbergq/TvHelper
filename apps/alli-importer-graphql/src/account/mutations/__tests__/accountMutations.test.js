// @flow

import { generateTestsFromFixtures } from '@adeira/test-utils';
import path from 'path';
import { UserRepository, alliImporterConnection } from '@tbergq/alli-importer-persistence';

import executeTestQuery from '../../../common/executeTestQuery';

beforeEach(async () => {
  await UserRepository.createUser({ email: 'test@test.com', password: '123' });
});
afterEach(async () => {
  await alliImporterConnection.collection('users').drop();
});

describe('accountMutations', () => {
  generateTestsFromFixtures(path.join(__dirname, '__fixtures__'), input => executeTestQuery(input));
});
