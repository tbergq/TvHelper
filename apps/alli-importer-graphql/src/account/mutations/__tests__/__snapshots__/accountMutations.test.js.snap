// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accountMutations matches expected output: createUser.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
mutation createUserMutation {
  createUser(user: {email:"lol@lol.no", password:"test"}) {
    __typename
    ... on User {
      id
      email
    }
    ... on GeneralError {
      message
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "createUser": {
      "__typename": "User",
      "id": "VXNlcjpsb2xAbG9sLm5v",
      "email": "lol@lol.no"
    }
  }
}
`;

exports[`accountMutations matches expected output: createUserError.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
mutation createUserMutation {
  createUser(user: {email:"test@test.com", password:"test"}) {
    __typename
    ... on User {
      id
      email
    }
    ... on GeneralError {
      message
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "createUser": {
      "__typename": "GeneralError",
      "message": "E11000 duplicate key error dup key: { : \\"test@test.com\\" }"
    }
  }
}
`;
